### ISSUE TITLE
Implement oauth support in the backend.

### ISSUE THREAD

**Issue by bmccandless** [chanzuckerberg/corpora-data-portal#481]
See [Authentication in Data Portal and Cellxgene](https://docs.google.com/document/d/19cC_tH7DOHIYqojdo-6T9eTzPmoMkwrAj83Ozvqth8s/edit#heading=h.t849jh3lwzxu) for details.

**TEAM COMMENTS**
1. If you point browser at the login endpoint and will take you to universal login page, then once you're redirected back to Data Portal check response header for cookies - Ask BrianM

**Comment by brianraymor** [comment]
Moving from Milestone 2 to Milestone 1 per the plan to _Document and implement AuthN/AuthZ infrastructure_ to unblock Milestone 2 features like user registration and user annotation.

**Comment by tihuan** [comment]
Hi @brianraymor ! Not sure if you're currently working on this, since it's in "Next" column. But can you please help estimate this story by noon today? (roughly 1 point === 1 day)

BrianR needs me to provide an estimate when the eng team can deliver M1, so I'm trying to get estimates for all the issues by noon and provide a rough delivery schedule 😇 Thanks a lot!

**Comment by tihuan** [comment]
@bmccandless I'm pointing it as a 5 for now. But please repoint it if you think it's not accurate and let me know! Thank you!

**Comment by bmccandless** [comment]
5 sounds good.

**Comment by tihuan** [comment]
Awesome thanks, Brian 🎉 !

**Comment by tihuan** [comment]
I estimated this with the following assumptions:

1. The issue will create a set of AuthN related API endpoints for FE to hit
2. BE will integrate with Auth0's tenant
3. We will use the default Universal Login design (And there will be a separate issue to skin the login/sign up page per Pat's design?)

Questions:
1. Is this ticket going to do the integration for all environments or just dev?
2. Where are we storing the environment config? Just wanted to make sure it will work for all environments / deployments

**Comment by bmccandless** [comment]
@tihuan, I agree with the assumptions

1.  The implementation in this is designed to work for all environments, not just dev.  

2.  The config is store in the AWS secret manager.   There are two key/values that real secrets, and a few items that are configuration.   Currently only the dev secret is defined, but staging and prod will be similar.

**Comment by bmccandless** [comment]
The first pass is complete:  basic login, logout and and endpoint to get userinfo.  Also tested on the dev deployment.

Todo:
 - handle refresh tokens
 - write a test case that exercises the flow
 - fix the redirects after login and logout.
